app/assets/javascripts/oxalis/model/reducers/skeletontracing_reducer_helpers.js

```diff
@@ -117,156 +117,215 @@ export function deleteNode(
-        activeTree.comments.forEach(comment => {
-          cutTrees.forEach(newTree => {
-            if (newTree.nodes[comment.node]) {
-              newTree.comments.push(comment);
-            }
-          });
-        });
-
-        newTrees = skeletonTracing.trees;
-        cutTrees.forEach(cutTree => {
-          newTrees = update(newTrees, { [cutTree.treeId]: { $set: cutTree } });
-        });
-        // newActiveNodeId was already written to when traversing the tree. Find the
-        // corresponding treeId
-        const newActiveTree = findTreeByNodeId(newTrees, newActiveNodeId).get();
-        newActiveTreeId = newActiveTree.treeId;
       }
 
-      // if the deleted node had the max id, find the new largest id
+      const newTrees = splitTreeByNodes(
+        state,
+        skeletonTracing,
+        activeTree,
+        neighborIds,
+        deletedEdges,
+        timestamp,
+      );
+
+      // If the deleted node had the max id, find the new largest id
+      let newMaxNodeId = skeletonTracing.cachedMaxNodeId;
       if (node.id === newMaxNodeId) {
         newMaxNodeId = getMaximumNodeId(newTrees);
       }
 
+      const newActiveNodeId = neighborIds[0];
+      const newActiveTree = findTreeByNodeId(newTrees, newActiveNodeId).get();
+      const newActiveTreeId = newActiveTree.treeId;
+
       return Maybe.Just([newTrees, newActiveTreeId, newActiveNodeId, newMaxNodeId]);
     } else {
       return Maybe.Nothing();
     }
   });
 }
 
+export function deleteEdge(
+  state: OxalisState,
+  sourceTree: TreeType,
+  sourceNode: NodeType,
+  targetTree: TreeType,
+  targetNode: NodeType,
+  timestamp: number,
+): Maybe<TreeMapType> {
+  return getSkeletonTracing(state.tracing).chain(skeletonTracing => {
+    const { allowUpdate } = skeletonTracing.restrictions;
+
+    if (allowUpdate) {
+      if (sourceTree.treeId !== targetTree.treeId) {
+        // The two selected nodes are in different trees
+        console.error(
+          "Tried two delete an edge that was not there, the two nodes are in different trees.",
+        );
+        return Maybe.Nothing();
+      }
+
+      console.log(sourceTree.edges, sourceNode.id, targetNode.id);
+      const deletedEdge = sourceTree.edges.find(
+        edge =>
+          (edge.source === sourceNode.id && edge.target === targetNode.id) ||
+          (edge.source === targetNode.id && edge.target === sourceNode.id),
+      );
+
+      if (deletedEdge == null) {
+        // The two selected nodes do not share an edge
+        console.error("Tried two delete an edge that was not there.");
+        return Maybe.Nothing();
+      }
+
+      return Maybe.Just(
+        splitTreeByNodes(
+          state,
+          skeletonTracing,
+          sourceTree,
+          [sourceNode.id, targetNode.id],
+          [deletedEdge],
+          timestamp,
+        ),
+      );
+    } else {
+      return Maybe.Nothing();
+    }
+  });
+}
+
+function splitTreeByNodes(
```

Comment:
Can you add a comment à la: "This function splits a given tree by deleting the given edges and making the given node ids new tree roots"?